doc/manualGrading.md

```diff
@@ -0,0 +1,56 @@
+# Manual Grading
+
+PrairieLearn supports manual grading of questions by downloading a CSV file with student answers and uploading a CSV file with question scores and optional per-question feedback. There is not currently an online web interface for manual grading.
+
+
+## Configuring a question for maual grading
+
+The [`info.json` file](question.md#question-infojson) in the question should be set for manual grading:
+```json
+{
+    "uuid": "cbf5cbf2-6458-4f13-a418-aa4d2b1093ff",
+    "gradingMethod": "Manual",
+    "singleVariant": true,
+    ...
+}
+```
+
+Manually-graded questions allow students to "Save" answers, but they don't have a "Save & Grade" button. Instead, the student just saves answers as many times as they want, and all of their submitted answers are stored. It is recommended to also mark manually-graded questions as `"singleVariant": true` so that students are only given a single random variant, even on Homework assessments.
+
+Any [elements](elements/) can be used in the [`question.html`](question.md#question-questionhtml) to write manually graded questions. All of the student input will be saved and available for manual grading, including `pl-string-input`, `pl-file-editor`, `pl-file-upload`, etc.
+
+To show manual feedback the `question.html` file should contain an element to display the feedback next to student submissions. A basic template for this is:
+```html
+<pl-submission-panel>
+  {{#feedback.manual}}
+  <p>Feedback from course staff:</p>
+  <markdown>{{{feedback.manual}}}</markdown>
+  {{/feedback.manual}}
+</pl-submission-panel>
+```
+
+This example template formats the feedback as Markdown.
+
+
+## Downloading the students' submitted answers
+
+After students have completed the assessment, download the submitted answers by going to the assessment page, then the "Downloads" tab, and selecting the `<assessment>_submissions_for_manual_grading.csv` file. This looks like:
+```csv
+uid,qid,submission_id,params,true_answer,submitted_answer,score_perc,feedback
+mwest@illinois.edu,explainMax,42983,{},{},{"ans": "returns the maximum value in the array"},,
+zilles@illinois.edu,explainMax,42984,{},{},{"ans": "gives the set of largest values in the object"},,
+zilles@illinois.edu,describeFibonacci,42987,{},{},{"ans": "calculates the n-th Fibonacci number"},,
+```
+
+This CSV file has two blank columns at the end, ready for the percentage score and optional feedback. The `submission_id` is an internal identifier that PrairieLearn uses to determine exactly which submitted answer is being graded. The `params` and `true_answer` columns show the question data.
```

Comment:
Instead of empty, which could be problematic.... How about:

1. Set all `score_perc` to 0 in the default CSV. 
2. Set a common message in `feedback` to `"Fill me in!"` or `"Great job on this question!"`

The latter can be dynamically changed per user with random adjectives if we want to mirror [`praise`](https://github.com/rladies/praise).

Reply:
Blank `score_perc` fields just don't set the score for that student, and blank `feedback` fields don't set feedback. I thought this was useful so that you could upload partially-complete grading CSV files?

I could add an extra column called `old_score_perc`, containing the current score? That way people could copy the old score if they are doing a regrade. On a first-ever grading the `old_score_perc` would be zero for all questions.

Reply:
For what it's worth, I prefer the blank fields.  For me, putting in an explicit 0 is different from not having graded it yet.  Updating a single grade is already pretty easy by just submitting a CSV with just that student so I can be absolutely sure I'm not messing up someone I already graded.

Reply:
I added the `old_score_perc` column, but I left the new columns blank.

Reply:
I also clarified in the docs that if you leave either `score_perc` or `feedback` (or both) blank for any student, then the corresponding entry will not be updated.